File: q0905/q0905.cpp
class Solution
{
public:
    vector<int> sortArrayByParity(vector<int> &nums)
    {
        int l = 0;
        int r = nums.size() - 1;
        int i = 0;
        while (l < r)
        {
            if (nums[i] % 2 == 0)
            {
                swap(nums[i], nums[l]);
                l++;
                i++;
            }
            else
            {
                swap(nums[i], nums[r]);
                r--;
            }
        }
        return nums;
    }
};

File: q0905/q0905.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int l = 0; 
        int r = nums.length - 1;
        int i = 0;
        while (l < r) {
            if (nums[i] % 2 == 0) {
                int temp = nums[l];
                nums[l] = nums[i];
                nums[i] = temp; 
                l++;
                i++;
            } else {
                int temp = nums[r];
                nums[r] = nums[i];
                nums[i] = temp; 
                r--;
            }
        } 
        return nums;
    }
}